### PR TITLE
add request-timeout to rt dial func

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://github.com/openshift/origin/issues/18159

cc @deads2k @liggitt @enj

### CODE REVIEW

File: pkg/oc/cli/cmd/login/loginoptions.go [hunk 1]
```diff
@@ -104,6 +104,13 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 	// ensure clientConfig has timeout option
 	if o.RequestTimeout > 0 {
 		clientConfig.Timeout = o.RequestTimeout
```
Comment (enj) [inline pkg/oc/cli/cmd/login/loginoptions.go]: Add comment that this timeout is used by the http client.

File: pkg/oc/cli/cmd/login/loginoptions.go [hunk 2]
```diff
@@ -104,6 +104,13 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
...
+
+		// if a timeout was specified, set a custom transport dial func that mimics the default one
+		// from k8s.io/client-go/transport/cache.go#get but with a custom timeout field.
+		clientConfig.Dial = (&net.Dialer{
+			Timeout:   o.RequestTimeout,
```
Comment (enj) [inline pkg/oc/cli/cmd/login/loginoptions.go]: And another comment saying that this is for the http round tripper.

**Comment by enj** [comment]
The "correct" solution would be to figure out some reasonable way to do this all upstream in client-go.  But since this timeout flag is part of origin code this seems OK for now.  I will defer LGTM to another reviewer.

/approve

File: pkg/oc/cli/cmd/login/loginoptions.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/oc/cli/cmd/login/loginoptions.go]: So, I took a look at https://github.com/kubernetes/client-go/blob/master/transport/cache.go#L79 which is where the dialer.Dial func ends up being used. Apparently it defaults to the http package's DefaultTransport, [which makes use of a dialer with a 30 second timeout](https://github.com/golang/go/blob/master/src/net/http/transport.go#L42)

**Pull request closed without merging**